librz/core/canalysis.c

```diff
@@ -5755,6 +5755,98 @@ RZ_IPI bool rz_core_analysis_var_rename(RzCore *core, const char *name, const ch
 	return true;
 }
 
+static bool is_in_data_section(RzCore *core, const ut64 address) {
+	if (address < 256 || address == UT64_MAX) {
+		return false;
+	}
+	RzBinObject *o = rz_bin_cur_object(core->bin);
+	const RzBinSection *section = o ? rz_bin_get_section_at(o, address, core->io->va) : NULL;
+	return section && section->vsize > 0 && strstr(section->name, "data");
```

Comment:
what if we rely on segment instead and check for r-- or rw- permissions?

Reply:
because is not reliable. there are bins with data that has the EXEC flag.

Reply:
Sections are even less reliable considering that in binaries like ELF they can even be removed completely without affecting the program execution at all. 

I think we can rely on both sections and segments. If the address falls in a section with data name, consider it data. Otherwise look at segment and their permissions.

Reply:
We can consider both Rx and rw and r as data. Rwx means self modifying code and who knows what happens in that case.

Reply:
hmm, ok agreed.